### PR TITLE
feat(Projects): enabled Project/Release mainline state change only for clearing admins

### PR DESCRIPTION
**Pull request by akapti**
Summary: Changes of the Project/Release Mainline State is only allowed for the role Clearing Admin.

**Steps to test:**

  **_Scenario 1:_**
```
1. Login with USER role.
2. Go to Projects tab and edit a Project which is having linked releases.
3. Select "Linked Releases And Projects" on left.
4. Verify that "Project Mainline State" drop-down is disabled. 
5. Click on "click to add Release" -> search for a release and add it to the "Linked Releases".
6. Verify that "Project Mainline State" drop-down is disabled. 
```

**Result:**
`1. User should not be able to change the "Project Mainline State" drop-down. in "Linked release and projects" tab.`

**_Scenario 2:_**
```
1. Login with USER role.
2. Go to Components tab and edit a component which is having release ("Release Overview" tab).
3. Select any release under that component.
4. Verify that "Release Mainline State" drop-down is disabled in "summary" tab. 
```
**Result:**
`1. User should not be able to change the "Release Mainline State" drop-down in "summary" tab.`

_**Verify the same for ADMIN and/or CLEARING_ADMIN**_
`Also, repeat the above mentioned test scenario 1 and 2 by logging in with ADMIN or CLEARING_ADMIN role, and Admin should have ability/permission to change the Project/Release Main State drop-down status.`

closes : #503 and #280 
Signed-off-by: Abdul Kapti <abdul.mannankapti@siemens.com>

### CODE REVIEW

**Comment by smrutis1** [comment]
**Testing Details: (Tested in Chrome browser)**

I have tested the PR. The functionality is working as expected. But there are few remarks regarding UI.
* The fields (release relations, project mainline state & delete icon) are overlapping with each other. I am wondering, if this can be addressed.



image

**Comment by akapti** [comment]
@smrutis1 I am not able to reproduce the UI issue in chrome and Firefox.
attached screenshot for reference:

**Chrome Browser:**
Chrome

**Firefox Browser:**
Firefox

File: frontend/sw360-portlet/src/main/webapp/html/components/editRelease.jsp [hunk 1]
```diff
@@ -43,6 +43,7 @@
 <liferay-theme:defineObjects/>
 
 <c:catch var="attributeNotFoundException">
+    <jsp:useBean id="isUserAtLeastClearingAdmin" class="java.lang.String" scope="request" />
```
Comment (imaykay) [inline frontend/sw360-portlet/src/main/webapp/html/components/editRelease.jsp]: It should be possible to have Boolean-beans like on `/sw360-portlet/src/main/webapp/html/components/detail.jsp` the bean `<jsp:useBean id="isUserAllowedToMerge" type="java.lang.Boolean" scope="request"/>`.

Comment (akapti) [reply frontend/sw360-portlet/src/main/webapp/html/components/editRelease.jsp]: @maxhbr and @imaykay 
Changed `isUserAtLeastClearingAdmin` to Boolean bean.
Added back-end validation for Mainline State change through web UI and REST interface as well.

**Comment by mcjaeger** [comment]
(from discussion) maybe not everyone woul dbe Ok with this restriction, how about making it configurable in sw360 properties? (ref #280)

**Comment by akapti** [comment]
Resolved merge conflicts, and made this feature configurable in sw360.properties.
Please review.

**Comment by nutanv1contr** [comment]
Is it possible to write a common permission check logic for all the request from different point of application and rest end point ?  Can be tried to make it centralized at  `ProjectDatabaseHandler.java` .

**Comment by akapti** [comment]
@nutanv1contr : Sounds good. I will check on that.

**Comment by nutanv1contr** [comment]
Tested the PR .Functionality is working as expected. Below are the scenarios test.

1. Create new project with linked releases by user role. "Project Mainline State" is disabled. Tried the same for "Admin" role , "Project Mainline State" is enabled for admin role. 
2. "Project Mainline State" is enable/disabled for user role as per the configuration in properties file.
3. Tested the feature from REST end point also. It is behaving as expected.

**Comment by nutanv1contr** [comment]
Code looks good.

**Pull request merged**